### PR TITLE
[5.8] skip testCompilationDiagnostics

### PR DESCRIPTION
**Pull request by tomerd**
motivation: stable CI

changes: disable testCompilationDiagnostics test since 5.8 has mismatched compiler

### CODE REVIEW

**Review by neonichu — APPROVED** [review]

**Comment by tomerd** [comment]
@compnerd any insight to the windows failure? seems unrelated

**Comment by compnerd** [comment]
Looks like a compiler crash?

```
tools\swift\lib\SwiftRemoteInspection\CMakeFiles\swiftRemoteInspection.dir\__\__\stdlib\public\RemoteInspection\TypeRefBuilder.cpp.obj
FAILED: tools/swift/lib/SwiftRemoteInspection/CMakeFiles/swiftRemoteInspection.dir/__/__/stdlib/public/RemoteInspection/TypeRefBuilder.cpp.obj 
C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1429~1.300\bin\Hostx64\x64\cl.exe  /nologo /TP -DGTEST_HAS_RTTI=0 -DLLVM_ON_WIN32 -DSWIFT_THREADING_WIN32 -DUNICODE -D_CRT_NONSTDC_NO_DEPRECATE -D_CRT_NONSTDC_NO_WARNINGS -D_CRT_SECURE_NO_DEPRECATE -D_CRT_SECURE_NO_WARNINGS -D_CRT_USE_WINAPI_FAMILY_DESKTOP_APP -D_DLL -D_ENABLE_ATOMIC_ALIGNMENT_FIX -D_ENABLE_EXTENDED_ALIGNED_STORAGE=1 -D_HAS_EXCEPTIONS=0 -D_HAS_STATIC_RTTI=0 -D_MD -D_SCL_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_WARNINGS -D_UNICODE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Itools\swift\lib\SwiftRemoteInspection -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\lib\SwiftRemoteInspection -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\stdlib\public\SwiftShims -Itools\swift\include -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include -Iinclude -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\llvm-project\llvm\include -Itools\clang\include -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\llvm-project\llvm\..\clang\include -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\cmark\src\include -Itools\cmark\src -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift-corelibs-libdispatch\src\BlocksRuntime -IC:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift-corelibs-libdispatch -I"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30037\\include" -I"C:\Program Files (x86)\Windows Kits\10\\Include\10.0.17763.0\ucrt" -I"C:\Program Files (x86)\Windows Kits\10\\Include\10.0.17763.0\shared" -I"C:\Program Files (x86)\Windows Kits\10\\Include\10.0.17763.0\um" /GS- /Oy /Gw /Gy /Zc:inline /Zc:__cplusplus /Zc:strictStrings /Oi /Zc:rvalueCast /bigobj /W4 -wd4141 -wd4146 -wd4244 -wd4267 -wd4291 -wd4351 -wd4456 -wd4457 -wd4458 -wd4459 -wd4503 -wd4624 -wd4722 -wd4100 -wd4127 -wd4512 -wd4505 -wd4610 -wd4510 -wd4702 -wd4245 -wd4706 -wd4310 -wd4701 -wd4703 -wd4389 -wd4611 -wd4805 -wd4204 -wd4577 -wd4091 -wd4592 -wd4319 -wd4709 -wd4324 -w14062 -we4238 /Gw /we4062 /wd4068 /permissive- -DOBJC_OLD_DISPATCH_PROTOTYPES=0 /MD /O2 /Ob2  /EHs-c- /GR- -UNDEBUG -O2 /GR- -U_DEBUG -DSWIFT_ENABLE_REFLECTION -std:c++17 /showIncludes /Fotools\swift\lib\SwiftRemoteInspection\CMakeFiles\swiftRemoteInspection.dir\__\__\stdlib\public\RemoteInspection\TypeRefBuilder.cpp.obj /Fdtools\swift\lib\SwiftRemoteInspection\CMakeFiles\swiftRemoteInspection.dir\swiftRemoteInspection.pdb /FS -c C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\stdlib\public\RemoteInspection\TypeRefBuilder.cpp
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include\swift/RemoteInspection/Records.h(383): warning C4200: nonstandard extension used: zero-sized array in struct/union
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include\swift/RemoteInspection/Records.h(383): note: This member will be ignored by a defaulted constructor or copy/move assignment operator
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include\swift/Runtime/Metadata.h(634): warning C4190: 'swift_initClassMetadata2' has C-linkage specified, but returns UDT 'swift::MetadataDependency' which is incompatible with C
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include\swift/ABI/Metadata.h(97): note: see declaration of 'swift::MetadataDependency'
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include\swift/Runtime/Portability.h(66): warning C4714: function 'int swift_asprintf(char **,const char *,...)' marked as __forceinline not inlined
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\include\swift/Runtime/Portability.h(66): note: see declaration of 'swift_asprintf'
C:\Users\swift-ci\jenkins\workspace\swiftpm-PR-windows\swift\stdlib\public\RemoteInspection\TypeRefBuilder.cpp(705) : fatal error C1001: Internal compiler error.
(compiler file 'd:\a01\_work\3\s\src\vctools\Compiler\Utc\src\p2\main.c', line 213)
 To work around this problem, try simplifying or changing the program near the locations listed above.
If possible please provide a repro here: https://developercommunity.visualstudio.com/ 
Please choose the Technical Support command on the Visual C++ 
 Help menu, or open the Technical Support help file for more information
  cl!RaiseException()+0x69
  cl!RaiseException()+0x69
  cl!InvokeCompilerPassW()+0x11c5af
  cl!InvokeCompilerPassW()+0x17aa3b
```

Not sure how that snuck in, we should double check what is the cause of that ... that seems bad.

**Comment by tomerd** [comment]
@swift-ci smoke test windows

**Comment by tomerd** [comment]
@compnerd should we merge despite the windows crash or do you have a path to fix this? we need to unblock landing fixes to 5.8

**Comment by compnerd** [comment]
> 
@compnerd should we merge despite the windows crash or do you have a path to fix this? we need to unblock landing fixes to 5.8

We need to fix this; I can take a look at this today (just in a little bit).

**Comment by compnerd** [comment]
This is interesting, I don't see any changes to the branch in the past 2 weeks?

**Comment by tomerd** [comment]
cc @shahmishal @buttaface this PR is to unblock https://github.com/apple/swift-package-manager/pull/6430 but now it itself is blocked on the windows issue

**Comment by compnerd** [comment]
I _think_ that I may have a workaround for this - it's the template metaprogramming in `TaggedUnionBase`.

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift/pull/65308

@swift-ci please test Windows platform

**Pull request merged**
